instrumentation/jmx-metrics/javaagent/camel.md

```diff
@@ -0,0 +1,54 @@
+| org.apache.camel.processor.external.redeliveries.count | Counter | context, route, processor, destination | External Redeliveries grouped by Camel Processor                                     |
+| org.apache.camel.processor.max.processing.time         | Gauge   | context, route, processor, destination | Maximum Processing Time grouped by Camel Processor                                   |
+| org.apache.camel.processor.mean.processing.time        | Gauge   | context, route, processor, destination | Mean Processing Time grouped by Camel Processor                                      |
+| org.apache.camel.processor.min.processing.time         | Gauge   | context, route, processor, destination | Min Processing Time grouped by Camel Processor                                       |
+| org.apache.camel.processor.last.processing.time        | Gauge   | context, route, processor, destination | Last Processing Time grouped by Camel Processor                                      |
+| org.apache.camel.processor.delta.processing.time       | Gauge   | context, route, processor, destination | Delta Processing Time grouped by Camel Processor                                     |
+| org.apache.camel.processor.total.processing.time       | Gauge   | context, route, processor, destination | Total Processing Time grouped by Camel Processor                                     |
+| org.apache.camel.processor.redeliveries.count          | Counter | context, route, processor, destination | Redeliveries count grouped by Camel Processor                                        |
+| org.apache.camel.processor.inflight.exchanges          | Gauge   | context, route, processor, destination | Inflight Exchanges grouped by Camel Processor                                        |
+| org.apache.camel.threadpool.active.count               | Gauge   | context, route                         | The approximate number of threads that are actively executing tasks.                                        |
+| org.apache.camel.threadpool.completed.task.count       | Counter | context, route                         | The approximate total number of tasks that have completed execution. Because the states of tasks and threads may change dynamically during computation, the returned value is only an approximation, but one that does not ever decrease across successive calls.                                        |
+| org.apache.camel.threadpool.pool.size                  | Gauge   | context, route                         | The current number of threads in the pool.                                        |
+| org.apache.camel.threadpool.largest.pool.size          | Gauge   | context, route                         | The largest number of threads that have ever simultaneously been in the pool.      |
```

Comment:
since it is monotonic I guess it should be a counter

Reply:
I respectfully disagree. The fact that it is monotonic does not make it a Counter. Counters need to "be additive", meaning that it makes sense to add Counter values together across its metric attributes (even if there might be non in that case).